src/ModelInterface.py

```diff
@@ -0,0 +1,56 @@
+
+
+
+import torch
+from torch import nn
+from torch import optim
+from torchvision import datasets, transforms
+from torch.utils.data import random_split, DataLoader
+import pytorch_lightning as pl
+import torchmetrics
+from typing import Optional
+from pytorch_lightning.loggers import CSVLogger
+from importlib import import_module
+
+
+class ModelInterface(pl.LightningModule):
+    def __init__(self, model_name):
+        super().__init__()
+        self.save_hyperparameters()
+        self.load_model(model_name)
+        self.lr = 1e-2
+
+    def load_model(self, name):
+        try:
+            Model = getattr(import_module(
+                'model.' + name), name)
+        except:
+            raise ValueError(
+                f'Invalid Module File Name {name}!')
+        self.model = Model()
+
+    def configure_optimizers(self):
```

Comment:
This is ok, but there are so many other optimizers. Sometimes I even want to implement my own optimizer. How do I handle the case where my optimizer is not SGD? Also you are hardcoding only the learning rate. What about weight decay? momentum? Finally, what if I want to have my own learning rate schedule?

Reply:
I used 'optimizer_func' to enable changes in optimizers. However, I found that I can't add more arguments such as weight decay. I tried to just use instantiated optimizer, but I couldn't provide the parameters.
For learning rate, I finally used a variable, but I didn't find some ways to enable learning rate schedulers.